bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerStationHandler.java

```diff
@@ -116,6 +117,33 @@ protected State getStationState(int stationId) {
         }
     }
 
+    /**
+     * Handles determining a channel's current state from the OpenSprinkler device.
+     *
+     * @param stationId Int of the station to control. Starts at 0.
+     * @return State representation for the channel.
+     */
+    @Nullable
+    protected State getRemainingWaterTime(int stationId) {
+        long remainingWaterTime = 0;
+        OpenSprinklerApi api = getApi();
+        if (api == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED,
+                    "OpenSprinkler bridge has no initialized API.");
+            return null;
+        }
+
+        try {
+            remainingWaterTime = api.retrieveProgram(stationId).remainingWaterTime;
+        } catch (Exception exp) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not get current state of station channel " + stationId
+                            + " for the OpenSprinkler device. Exception received: " + exp);
+        }
+
+        return new DecimalType(remainingWaterTime);
```

Comment:
```suggestion
        return new QuantityType<Time>(remainingWaterTime, SIUnits.SECOND);
```

May not be 100% correct, please check

Reply:
Done!